publish-subscribe/README.md

```diff
@@ -0,0 +1,237 @@
+---
+title: "Publish-Subscribe Pattern in Java: Decoupling the solution with asynchronous communication"
+shortTitle: Proxy
+description: "Explore the Proxy design pattern in Java with detailed examples. Learn how it provides controlled access, facilitates lazy initialization, and ensures security. Ideal for developers looking to implement advanced Java techniques."
+category: Structural
+language: en
+tag:
+    - Decoupling
+    - Encapsulation
+    - Gang Of Four
+    - Lazy initialization
+    - Proxy
+    - Security
+    - Wrapping
+---
+
+## Intent of the Publish-Subscribe Design Pattern
+
+The Publish-Subscriber design pattern is widely used in software architecture to transmit data between various components in a system.
+It is a behavioral design pattern aimed at achieving loosely coupled communication between objects.
+The primary intent is to allow a one-to-many dependency relationship where one object (the Publisher) notifies multiple other objects (the Subscribers) about changes or events,
+without needing to know who or what the subscribers are.
+
+## Detailed Explanation of Publish-Subscribe Pattern with Real-World Examples
+
+- Messaging systems like Kafka, RabbitMQ, AWS SNS, JMS
+    - **Kafka** : publishes messages to topics and subscribers consumes them in real time for analytics, logs or other purposes.
+    - **RabbitMQ** : Uses exchanges as publisher and queues as subscribers to route messages
+    - **AWS SNS** : Simple Notification Service (SNS) received the messages from publishers with topic and the subscribers on that topic will receive the messages. (SQS, Lambda functions, emails, SMS)
+
+
+- Event driven microservices
+    - **Publisher** : Point of Sale(PoS) system records the sale of an item and publish the event
+    - **Subscribers** : Inventory management service updates stock, Billing service sends e-bill to customer
+
+
+- Newsletter subscriptions
+    - **Publisher** : Writes a new blog post and publish to subscribers
+    - **Subscribers** : All the subscribers to the newsletter receive the email
```

Comment:
See the template for required contents of this section: https://github.com/iluwatar/java-design-patterns/wiki/21.-New-Pattern-Template